spec/services/healthcheck_spec.rb

```diff
@@ -0,0 +1,22 @@
+require 'rails_helper'
+
+RSpec.describe Healthcheck, type: :service do
+  describe '.perform' do
+    it 'returns an hash with the healthy status' do
+      result = described_class.perform
+      expect(result).to eq(database: true)
+    end
+
+    context 'when the database is not healthy' do
+      let(:mocked_connection) { double(:active_record_connection) }
```

Comment:
There is a mantra of "[Don't mock what you don't own](https://8thlight.com/blog/eric-smith/2011/10/27/thats-not-yours.html)" which applies in this case.

In this example, if the rails connection class broke the `active?` method during an upgrade, your test suite wouldn't notice.

I'd recommend investigating if using the `disconnect!` method would allow you to test the behaviour without mocking out the `active?` method.